Is this project supposed to be built with gradle?

I see gradle files at the root, but I'm unable to build it. Attaching the complete logs. 



```
$ gradle --version 

------------------------------------------------------------
Gradle 4.4.1
------------------------------------------------------------

Build time:   2017-12-20 15:45:23 UTC
Revision:     10ed9dc355dc39f6307cc98fbd8cea314bdd381c

Groovy:       2.4.12
Ant:          Apache Ant(TM) version 1.9.9 compiled on February 2 2017
JVM:          1.8.0_144 (Oracle Corporation 25.144-b01)
OS:           Linux 4.14.13-1-ARCH amd64

$ gradle build 

:compileJavawarning: [options] bootstrap class path not set in conjunction with -source 1.7
warning: Supported source version 'RELEASE_6' from annotation processor 'jenkins.PluginSubtypeMarker' less than -source '1.7'
warning: No SupportedSourceVersion annotation found on org.kohsuke.wpc.ProcessorImpl, returning RELEASE_6.
warning: Supported source version 'RELEASE_6' from annotation processor 'org.kohsuke.wpc.ProcessorImpl' less than -source '1.7'
warning: Supported source version 'RELEASE_6' from annotation processor 'org.kohsuke.stapler.jsr269.ConstructorProcessor' less than -source '1.7'
warning: Supported source version 'RELEASE_6' from annotation processor 'org.kohsuke.stapler.jsr269.ExportedBeanAnnotationProcessor' less than -source '1.7'
warning: Supported source version 'RELEASE_6' from annotation processor 'org.kohsuke.stapler.jsr269.QueryParameterAnnotationProcessor' less than -source '1.7'
warning: Supported source version 'RELEASE_6' from annotation processor 'org.kohsuke.stapler.jsr269.RequirePOSTAnnotationProcessor' less than -source '1.7'
warning: Supported source version 'RELEASE_6' from annotation processor 'org.kohsuke.stapler.jsr269.WebMethodAnnotationProcessor' less than -source '1.7'
warning: Supported source version 'RELEASE_6' from annotation processor 'org.jvnet.hudson.annotation_indexer.AnnotationProcessorImpl' less than -source '1.7'
warning: Supported source version 'RELEASE_6' from annotation processor 'net.java.sezpoz.impl.Indexer6' less than -source '1.7'
Note: org.jenkinsci.plugins.prometheus.rest.PrometheusAction indexed under hudson.Extension
Note: org.jenkinsci.plugins.prometheus.config.PrometheusConfiguration indexed under hudson.Extension
/home/j/Projects/prometheus-plugin/src/main/java/org/jenkinsci/plugins/prometheus/JobCollector.java:21: error: package hudson.tasks.test does not exist
import hudson.tasks.test.AbstractTestResultAction;
                        ^
1 error
10 warnings
 FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 0s
1 actionable task: 1 executed
```

--- a/build.gradle
+++ b/build.gradle
@@ -2,6 +2,7 @@ plugins{
   id 'java'
   id 'maven'
   id 'eclipse'
+  id 'org.jenkins-ci.jpi' version '0.18.0'
 }
 
 group = 'org.jenkins-ci.plugins'
@@ -17,19 +18,41 @@ repositories {
      maven { url "http://repo.maven.apache.org/maven2" }
 }
 
+// Only compile properties with with Java compiler.
+// This is for Messages.properties
+sourceSets.main.java.srcDirs += [ "src/main/resources/*.properties" ]
+
+// Make sure that generated Java files get compiled by the Java compiler
+// This is for Messages.java
+sourceSets.main.java.srcDirs += [ "build/generated-src/localizer" ]
+
+jenkinsPlugin {
+    // version of Jenkins core this plugin depends on, must be 1.420 or later
+    coreVersion = '1.625.3'
+    shortName = 'prometheus'
+    displayName = 'Prometheus plugin'
+    url = 'https://plugins.jenkins.io/prometheus'
+    gitHubUrl = 'https://github.com/jenkinsci/prometheus-plugin.git'
+
+    // use the plugin class loader before the core class loader
+    pluginFirstClassLoader = true
+}
+
 dependencies {
     compile 'org.jenkins-ci.main:jenkins-core:1.625.3',
     'org.jenkins-ci.plugins.workflow:workflow-api:1.8@jar',
     'org.jenkins-ci.plugins.workflow:workflow-job:1.8@jar',
     'org.apache.httpcomponents:httpcore:4.4',
-    'org.jenkins-ci.plugins:metrics:3.1.2.7@jar',
     'com.google.code.gson:gson:2.3.1',
     'io.prometheus:simpleclient:0.0.14',
     'io.prometheus:simpleclient_common:0.0.14',
     'io.prometheus:simpleclient_dropwizard:0.0.14',
     'javax.servlet:servlet-api:2.4',
     'org.slf4j:slf4j-api:1.7.7'
-    
+
+    jenkinsPlugins 'org.jenkins-ci.plugins:junit:1.21@jar',
+            'org.jenkins-ci.plugins:metrics:3.1.2.7@jar'
+
     testCompile 'org.jenkins-ci.main:jenkins-war:1.625.3:war-for-test',
     'org.jenkins-ci.main:jenkins-war:1.625.3',
     'org.jenkins-ci.main:jenkins-test-harness:2.1',
@@ -41,7 +64,7 @@ dependencies {
 }
 
 task dumpDependencies(type: Copy) {
-  description "Copy all project dependencies into ${buildDir}/dependencyLi    bs"
+  description "Copy all project dependencies into ${buildDir}/dependencyLibs"
   into "$buildDir/dependencyLibs"
   //include all dependencies, even test
   from configurations.testRuntime

--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -1,6 +1,6 @@
-#Tue Jun 14 13:35:40 EDT 2016
+#Wed Apr 25 08:09:44 CDT 2018
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.14-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.6-bin.zip

